src/app/screens/connect/btcSelectAddressScreen/index.tsx

```diff
@@ -0,0 +1,292 @@
+  const [appIcon, setAppIcon] = useState<string>('');
+  const { payload, origin, approveBtcAddressRequest, cancelAddressRequest } =
+    useBtcAddressRequest();
+  const appUrl = useMemo(() => origin.replace(/(^\w+:|^)\/\//, ''), [origin]);
+
+  const styles = useSpring({
+    from: {
+      opacity: 0,
+      y: 24,
+    },
+    to: {
+      y: 0,
+      opacity: 1,
+    },
+  });
+
+  const confirmCallback = async () => {
+    setLoading(true);
+    approveBtcAddressRequest();
+    window.close();
+  };
+
+  const cancelCallback = () => {
+    cancelAddressRequest();
+    window.close();
+  };
+
+  const switchAccountBasedOnRequest = () => {
+    if (payload.network.type !== network.type) {
+      navigate('/tx-status', {
+        state: {
+          txid: '',
+          currency: 'BTC',
+          errorTitle: t('NETWORK_MISMATCH_ERROR_TITLE'),
+          error: t('NETWORK_MISMATCH_ERROR_DESCRIPTION'),
+          browserTx: true,
+        },
+      });
+    }
+  };
+
+  useEffect(() => {
+    switchAccountBasedOnRequest();
+  }, []);
+
+  useEffect(() => {
+    (async () => {
+      if (origin !== '') {
+        getAppIconFromWebManifest(origin).then((appIcons) => {
+          setAppIcon(appIcons);
+        });
+      }
+    })();
+
+    return () => {
+      setAppIcon('');
+    };
+  }, [origin]);
+
+  const AddressPurposeRow = useCallback((purpose: AddressPurpose) => {
+    if (purpose === AddressPurpose.Payment) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={BitcoinIcon} />
+            <AddressTextTitle>{t('BITCOIN_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <TruncatedAddress>{getTruncatedAddress(btcAddress)}</TruncatedAddress>
+        </AddressBox>
+      );
+    }
+    if (purpose === AddressPurpose.Ordinals) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={OrdinalsIcon} />
+            <AddressTextTitle>{t('ORDINAL_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <TruncatedAddress>{getTruncatedAddress(ordinalsAddress)}</TruncatedAddress>
+        </AddressBox>
+      );
+    }
+    if (purpose === AddressPurpose.Stacks) {
+      return (
+        <AddressBox key={purpose}>
+          <AddressContainer>
+            <AddressImage src={stxIcon} />
+            <AddressTextTitle>{t('STX_ADDRESS')}</AddressTextTitle>
+          </AddressContainer>
+          <div>
+            {selectedAccount?.bnsName ? <BnsName>{selectedAccount?.bnsName}</BnsName> : null}
+            <TruncatedAddress>{getTruncatedAddress(stxAddress)}</TruncatedAddress>
+          </div>
+        </AddressBox>
+      );
+    }
+
+    navigate('/tx-status', {
+      state: {
+        txid: '',
+        currency: 'BTC',
+        errorTitle: t('INVALID_PURPOSE_ERROR_TITLE'),
+        error: t('INVALID_PURPOSE_ERROR_DESCRIPTION'),
+        browserTx: true,
+      },
+    });
+  }, []);
```

Comment:
why is this in a useCallback?

these AddressPurpose components won't react to any state changes to selectedAccount, stxAddress, stxIcon, ordinalsAddress, etc. 

it would be better to create a new component for <AddressPurpose> and pass in props